app/routes/preprints/provider.js

```diff
@@ -13,6 +13,7 @@ export default Route.extend({
     theme: service(),
 
     model(params) {
+        // need to load the provider here so theme can be used in child routes
         return this.get('theme').loadProvider(params.provider_id)
```

Comment:
I'm okay with loading the provider in the route, but could we do it without using `theme`? And pass the provider into the components that need it?

A nice thing about that is we could link to this page from the dashboard without the route hooks running, since we already have the provider with counts and everything.

Reply:
The issue with the provider is that it gets overwritten in the store all the time. So if there is a request to the `preprint_provider` endpoint that doesn't get the review counts then all of a sudden the counts aren't there.

Reply:
If it's consistently loaded in the route and/or dashboard (both with counts), and we're careful not to reload it otherwise, won't the store continue to reuse the same provider (with counts)?

...I guess even if that's true, it's fragile and requires that we remember to give providers special treatment. What if we extend the provider adapter to add `meta[reviews_state_counts]=true` to all requests, so we don't need to think about it elsewhere?

Reply:
The counts are in different locations too. The request on the dashboard page and the request on the provider page overwrite each other in the store. I think it would require backend changes (although probably a good idea). Below are the provider requests.

```js
const results = yield this.get('store').findRecord('preprint-provider', this.get('provider.id'), {
    adapterOptions: {
        query: { related_counts: true },
    },
    reload: true,
});
this.set('pendingCount', results.get('reviewableStatusCounts.pending'));
```

I don't think there is way to get the counts here?
```js
const results = yield this.get('store').query('preprint-provider', query);
return this.set('providers', results);
```

The request for the provider here could probably be updated to include the counts.
```js
const provider = yield this.get('store').findRecord(
    'preprint-provider',
    this.get('theme.id'),
);
const response = yield this.get('store').queryHasMany(provider, 'preprints', {
    filter: {
        reviews_state: queryParams.status,
        node_is_public: true,
    },
    'meta[reviews_state_counts]': true,
    sort: queryParams.sort,
    page: queryParams.page,
});
this.set('results', {
    submissions: response.toArray(),
    totalPages: Math.ceil(response.meta.total / response.meta.per_page),
    statusCounts: response.meta.reviews_state_counts,
});
```

Reply:
How would you feel about loading the provider in a controller task (triggered by this route's `setupController`) and passing the provider down to the components that need it?

Reply:
If `theme.provider` is loaded in the provider controller then the provider can be accessed through the theme and doesn't need to be passed down. That way it would be consistent with `theme.reviewableStatusCounts`.

Reply:
Makes sense 👍

Reply:
So I remembered why this doesn't work. The dynamic path is only passed to the model hook so even if setupController is used to call a task on the controller, the provider id isn't going to be available.

Reply:
Oh, and setupController gets the result of the model hook. I suppose `model` could just return the parameter that setupController needs, and call that the model, but that might be ugly.